Protocol.md

```diff
@@ -300,6 +317,173 @@ The recipient:
+
+## The `offer_channel` message
+
+1. type: 43000 (`offer_channel_v0`)
+2. data:
+   * [`u32`: `protocol_version`]
+   * [`byte`:`contract_flags`]
+   * [`chain_hash`:`chain_hash`]
+   * [`32*byte`:`temporary_contract_id`]
+   * [`32*byte`:`temporary_channel_id`]
+   * [`contract_info`:`contract_info`]
+   * [`point`:`funding_pubkey`]
+   * [`point`:`revocation_base_point`]
+   * [`point`:`publish_base_point`]
+   * [`point`:`own_base_point`]
+   * [`point`:`first_per_update_point`]
+   * [`spk`:`payout_spk`]
+   * [`u64`:`payout_serial_id`]
+   * [`u64`:`offer_collateral_satoshis`]
+   * [`bigsize`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`u64`:`change_serial_id`]
+   * [`u64`:`fund_output_serial_id`]
+   * [`u64`:`feerate_per_vb`]
+   * [`u32`:`cet_locktime`]
+   * [`u32`:`refund_locktime`]
+   * [`u32`:`cet_nsequence`]
+   * [`offer_channel_tlvs`: `tlvs`]
+
+## The `accept_channel` message
+
+1. type: 43002 (`accept_channel_v0`)
+2. data:
+   * [`32*byte`:`temporary_channel_id`]
+   * [`u64`:`accept_collateral_satoshis`]
+   * [`point`:`funding_pubkey`]
+   * [`point`:`revocation_base_point`]
+   * [`point`:`publish_base_point`]
+   * [`point`:`own_base_point`]
+   * [`point`:`first_per_update_point`]
+   * [`spk`:`payout_spk`]
+   * [`u64`:`payout_serial_id`]
+   * [`bigsize`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+   * [`spk`:`change_spk`]
+   * [`u64`:`change_serial_id`]
+   * [`cet_adaptor_signatures`:`cet_adaptor_signatures`]
+   * [`adaptor_signature`:`buffer_adaptor_signature`]
+   * [`signature`:`refund_signature`]
+   * [`negotiation_fields`:`negotiation_fields`] (Optional: 1)
+   * [`accept_channel_tlvs`: `tlvs`]
+
+## The `sign_channel` message
+
+1. type: 43004 (`sign_dlc_v0`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`cet_adaptor_signatures`:`cet_adaptor_signatures`]
+   * [`signature`:`refund_signature`]
+   * [`funding_signatures`:`funding_signatures`]
+   * [`sign_channel_tlvs`: `tlvs`]
+
+## The `settle_offer` message
+
+1. type: 43006
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u64`:`counter_payout`]
+   * [`point`: `next_per_update_point`]
+
+## The `settle_accept` message
+
+1. type: 43008
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`point`:`next_per_update_point`]
+   * [`adaptor_signature`:`settle_adaptor_signature`]
+
+## The `settle_confirm` message
+
+1. type: 43010
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`32*byte`:`prev_per_update_secret`]
+   * [`adaptor_signature`:`settle_adaptor_signature`]
+
+## The `settle_finalize` message
+
+1. type: 43012
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`32*byte`:`prev_per_update_secret`]
+
+
+## The `renew_offer` message
+
+1. type: 43014
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`contract_id`:`temporary_contract_id`]
+   * [`u64`:`counter_payout`]
```

Comment:
In our usecase of CFD-DLCs we have the concept of _funding fees_, i.e. a fee which needs to be paid to keep a position open. 
Do you intend to implicitly put this into `counter_payout`?

Reply:
We used to have something similar in our [p2pderivatives app](https://github.com/p2pderivatives/p2pderivatives-client), users could set a premium to pay to the other party which would be included in the funding transaction. Though in that case if it doesn't require a different output I think indeed it might be better to have it implicit in the `counter_payout`. You could then have some extra TLVs for application specific things like that (I haven't specified the extra TLV field yet but you can assume any message can have them).